### PR TITLE
fix OBJ mimes to make batch_importer happy. @dltj++ @adam-vessey

### PR DESCRIPTION
**Pull request by ruebot**
What the commit message says. I have test objects and a sample if needed.

### CODE REVIEW

File: xml/islandora_basic_image_ds_composite_model.xml [hunk 1]
```diff
@@ -22,6 +22,8 @@
   </dsTypeModel>
   <dsTypeModel ID="OBJ" ORDERED="false" optional="true">
```
Comment (nigelgbanks) [inline xml/islandora_basic_image_ds_composite_model.xml]: Lets catch all the instances of ORDERED!

https://wiki.duraspace.org/display/FCREPO/DSCOMPOSITE-MODEL+and+Schemas

**Comment by adam-vessey** [comment]
Just remembered something about this... The whole multiple datastreams which can have the same datastream ID...  "MEDIUM_SIZE" (if it had the proper mimetype) could collide with the definition for OBJ...  In some cases, we can end up not knowing which datastream ID to assign to the given content...  Just something to bear in mind?

**Comment by nigelgbanks** [comment]
I'm afraid I'm not sure what you mean Adam? Some section of code making decisions based on mimetype rather than ID?

**Comment by adam-vessey** [comment]
Yep. As the pull request title says, the big reason behind this was for the batch importer, which tries to bind datastream IDs based on MIME-type: https://github.com/Islandora/islandora_importer/blob/7.x/modules/zip_importer/includes/importer.inc#L267-L275 ...  Had been doing this initially, in https://github.com/Islandora/islandora_importer/blob/7.x/modules/zip_importer/includes/importer.inc#L215-L242...  For some reason, instead of uncommenting, somebody reimplemented...  Dunno.  Looks like it was done without trying to address the issue indicated in the comment in the determineDSID() method.

**Comment by adam-vessey** [comment]
One way to address it, might be to filter the things to match on down to the "required" datastreams (get rid of everything with `optional="true"`, just when mapping, I mean)?

**Comment by adam-vessey** [comment]
...  Or at least to weight them higher? Dunno

**Comment by nigelgbanks** [comment]
Oh ok this is coming back to me, my coding standards changes make the history a bit hard to read. Hmm could we just provide more information at the beginning of the batch? Like a mapping file or something, providing an Regex or something if the mime-type is not enough to determine the data-stream? Seems a bit like an intractable problem to do generically, without providing additional information at the outset? But maybe I'm missing something I'm not super familiar with the code.

**Pull request merged**